src/test/java/org/folio/rest/OpenTransactionsTest.java

```diff
@@ -0,0 +1,338 @@
+package org.folio.rest;
+
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.specification.RequestSpecification;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.folio.rest.tools.client.test.HttpClientMock2;
+import org.folio.rest.tools.utils.NetworkUtils;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.UUID;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+
+@RunWith(VertxUnitRunner.class)
+public class OpenTransactionsTest {
+
+  static Vertx vertx;
+  private static RequestSpecification okapi;
+  private static int okapiPort;
+  private static int port;
+
+  private static final String USER_ID = "0bb4f26d-e073-4f93-afbc-dcc24fd88810";
+  private static final String USER_BARCODE = "12345";
+  private static final String USER_NAME = "maxi";
+  private static final String LOAN_ID = "dfa12cc0-b82e-4554-9cb0-4f5bb1fdca01";
+  private static final String REQUEST_ID = "dfa12cc0-b82e-4554-9cb0-4f5bb1fdca02";
+  private static final String ACCOUNT_ID = "dfa12cc0-b82e-4554-9cb0-4f5bb1fdca03";
+  private static final String PROXY_ID_1 = "dfa12cc0-b82e-4554-9cb0-4f5bb1fdca04";
+  private static final String PROXY_ID_2 = "dfa12cc0-b82e-4554-9cb0-4f5bb1fdca05";
+  private static final String MANUAL_BLOCK_ID = "dfa12cc0-b82e-4554-9cb0-4f5bb1fdca06";
+
+  @BeforeClass
+  public static void before(TestContext context) {
+    vertx = Vertx.vertx();
+    vertx.exceptionHandler(context.exceptionHandler());
+
+    okapiPort = NetworkUtils.nextFreePort();
+    DeploymentOptions okapiOptions = new DeploymentOptions()
+      .setConfig(new JsonObject().put("http.port", okapiPort));
+    vertx.deployVerticle(MockOkapi.class.getName(), okapiOptions, context.asyncAssertSuccess());
+
+    port = NetworkUtils.nextFreePort();
+    DeploymentOptions usersBLOptions = new DeploymentOptions()
+      .setConfig(new JsonObject().put("http.port", port).putNull(HttpClientMock2.MOCK_MODE));
+    vertx.deployVerticle(RestVerticle.class.getName(), usersBLOptions, context.asyncAssertSuccess());
+
+    RestAssured.port = port;
+    RequestSpecBuilder builder = new RequestSpecBuilder();
+    builder.addHeader("X-Okapi-URL", "http://localhost:" + okapiPort);
+    builder.addHeader("X-Okapi-Tenant", "supertenant");
+    builder.addHeader("X-Okapi-Token", token("supertenant", "maxi"));
+    okapi = builder.build();
+    insertData();
```

Comment:
Race condition, insertData may start before verticle deployment has finished.
For details see
https://issues.folio.org/browse/MODUSERBL-118
https://github.com/folio-org/mod-users-bl/pull/117

Reply:
You're right. Thank you!